File: DownloadFiles/TGNet/Chapter05/Chapter5DemoCode/C#/SynchronousTiming/SynchronousTiming/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace SynchronousTiming
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            DateTime StartTime;
            DateTime EndTime;
            TimeSpan Interval;
            StartTime = DateTime.Now;
            Interaction.Shell
                ("C:\\Program Files\\Internet Explorer\\iexplore.exe", 
                AppWinStyle.NormalNoFocus, true, 5000);
           
            EndTime = DateTime.Now;
            Interval = EndTime - StartTime;
            MessageBox.Show(Interval.ToString());

        }
    }
}

File: DownloadFiles/TGNet/Chapter05/Chapter5DemoCode/VB/SynchronousTiming/SynchronousTiming/Form1.vb


Public Class Form1

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        Dim StartTime As DateTime
        Dim EndTime As DateTime
        Dim Interval As TimeSpan

        StartTime = DateTime.Now 'capture current seconds
        Shell("C:\Program Files\Internet Explorer\iexplore.exe", AppWinStyle.NormalFocus, True)
        EndTime = DateTime.Now

        Interval = EndTime - StartTime
        MessageBox.Show(Interval.ToString())
    End Sub
End Class
